changelogs/unreleased/gh-3682-uri-encode-decode.md

```diff
@@ -0,0 +1,3 @@
+## feature/lua/uri
```

Comment:
It is actually my request, but I would clarify my position here a bit for the record and to avoid misunderstanding in a future. All points are my opinion and can be argued.

* I don't consider the default unsafe character classes as something that a user should change (at least we shouldn't suggest it IMHO). So I wouldn't call it 'settings'. If a user needs his/her own unsafe set, he/she can (and should) pass it to the (un)escaping function. We agreed to discuss this point with Sergey, this is the open question.
* Strictly speaking, I didn't request this information to be in the documentation. I want the logic behind those character classes explained *somewhere* (my request is [here](https://github.com/tarantool/tarantool/pull/7254#discussion_r981743510)). I don't mind to place the result into the documentation if it'll look useful for a user. I'm a bit sad that Sergey declined my proposal to add this information as comments into the code (so one would able to just read and understand), but I'll not insist.
* It is not my caprice. It is the necessary work to make the classes correct and usable. Without this work we'll have what we have now: an incorrect implementation. See [here](https://github.com/tarantool/tarantool/pull/7254#discussion_r981795937) and [here](https://github.com/tarantool/tarantool/pull/7254#discussion_r981807580) for example.

Reply:
> Please stop spam in comments.

Wow, a bit offensive. We're working together, not against each other. I see nothing bad in clarification of my words.

Back to the topic (a description of the character classes/escape modes). The commit message/the documentation request mention them without much details, but there is an internal link for the team and it contains all the details. It is acceptable.

We all are a bit tired working on it, so I recall my point regarding comments on the character classes in the code.

Closing this topic.